www/docs/server/context.md

```diff
@@ -60,11 +62,88 @@ export const router = t.router;
 
 /**
  * Unprotected procedure
- **/
+ */
 export const publicProcedure = t.procedure;
 
 /**
  * Protected procedure
- **/
+ */
 export const protectedProcedure = t.procedure.use(isAuthed);
-```
\ No newline at end of file
+```
+
+## Inner and outer context
+
+In some scenarios it could make sense to split up your context into "inner" and "outer" functions.
+
+**Inner context** is where you define context which doesn’t depend on the request, e.g. your database connection. You can use this function for integration testing or [SSG helpers](ssg-helpers), where you don’t have a request object. Whatever is defined here will **always** be available in your procedures.
+
+**Outer context** is where you define context which depends on the request, e.g. for the user's session. Whatever is defined here is not available in all procedures, but only for procedures that are called via HTTP.
+
+### Example for inner & outer context
+
+```ts
+import type { inferAsyncReturnType } from '@trpc/server';
+import type { CreateNextContextOptions } from '@trpc/server/adapters/next';
+
+/** 
+ * Defines your inner context shape.
+ * Add fields here that the inner context brings.
+ */
+interface CreateInnerContextOptions extends Partial<CreateNextContextOptions> {
+  prisma: typeof prisma
+}
+
+/** 
+ * Inner context. Will always be available in your procedures, in contrast to the outer context.
+ * 
+ * Also useful for:
+ * - testing, so you don't have to mock Next.js' `req`/`res`
+ * - tRPC's `createSSGHelpers` where we don't have `req`/`res`
+ * 
+ * @see https://trpc.io/docs/context#inner-and-outer-context
+ */
+export async function createContextInner(opts?: CreateInnerContextOptions) {
+  return {
+    ...opts,
+    prisma,
+  }
+};
+
+/** 
+ * Outer context. Will e.g. bring `req` & `res` to the context as "not `undefined`".
+ * 
+ * @see https://trpc.io/docs/context#inner-and-outer-context
+ */
+export async function createContext(opts: CreateNextContextOptions) {
```

Comment:
Does this need to be async? I just realized we have this as async in create-t3-app too and I'm not sure why. Will try to figure this out.